x-pack/platform/plugins/shared/stack_connectors/server/connector_types/resilient/utils.ts

```diff
@@ -5,51 +5,127 @@
+import type {
+  ResilientUpdateFieldValue,
+  ResilientTextAreaField,
+  ResilientFieldPrimitives,
+  UpdateIncidentRequest,
+  CreateIncidentData,
+  Incident,
+} from './types';
+import type { ResilientFieldMeta } from './schema';
+
+const getValueFromOldField = (
+  fieldMeta: ResilientFieldMeta,
+  value: unknown
+): ResilientFieldPrimitives => {
+  switch (fieldMeta.input_type) {
+    case 'textarea':
+      return value === null || value === undefined
+        ? null
+        : (value as ResilientTextAreaField['textarea']);
+    case 'multiselect':
+    case 'select':
+    case 'datetimepicker':
+    case 'datepicker':
+    case 'boolean':
+    case 'number':
+    case 'text':
+      return value as ResilientFieldPrimitives;
+    default:
+      return null;
   }
+};
 
-  if (field === 'incidentTypes') {
-    if (isArray(value)) {
-      return { ids: value.map((item) => Number(item)) };
+export function validateValues(
+  fieldMeta: ResilientFieldMeta,
+  givenValues: ResilientFieldPrimitives[]
+) {
+  const valuesMeta = fieldMeta.values;
+  if (valuesMeta && Array.isArray(valuesMeta)) {
+    if (!givenValues.every((id) => valuesMeta.some((option) => option.value === id))) {
+      throw new Error(
+        `Invalid values provided to ${fieldMeta.name}: ${JSON.stringify(
+          givenValues,
+          null,
+          2
+        )}. Accepted values: ${valuesMeta.map((v) => `${v.value} (for "${v.label}")`)}`
+      );
     }
-    return {
-      ids: [Number(value)],
-    };
   }
+  return true;
+}
 
-  if (field === 'severityCode') {
-    return {
-      id: Number(value),
-    };
-  }
+function getValueFieldShape(
+  fieldMeta: ResilientFieldMeta,
+  value: ResilientFieldPrimitives,
+  oldValue?: ResilientFieldPrimitives
+): ResilientUpdateFieldValue {
+  switch (fieldMeta.input_type) {
+    case 'textarea':
+      if (value === null || value === undefined) {
+        return { textarea: null };
+      } else if (typeof value === 'object' && 'format' in value && 'content' in value) {
+        return {
+          textarea: {
+            ...value,
+          },
+        };
+      } else {
+        return {
+          textarea: {
+            format: isObject(oldValue) && 'format' in oldValue ? oldValue?.format : 'text',
+            content: value,
+          },
+        };
+      }
 
-  return {
-    text: value.toString(),
-  };
-};
+    case 'multiselect':
+      if (value === null || value === undefined) {
+        return {};
+      }
+      const ids = Array.isArray(value) ? value.map((item) => item) : [value];
+      if (validateValues(fieldMeta, ids)) {
+        return { ids };
+      }
+    case 'select':
+      if (value === null || value === undefined) {
+        return {};
+      }
+      if (validateValues(fieldMeta, [value])) {
+        return { id: value };
+      }
+    case 'datetimepicker':
+    case 'datepicker':
+      return isNumber(value) ? { date: value } : {};
```

Comment:
Just confirming this is always expected to be number and doesn't need an additional `isDate()` conversion/check?